server/build.gradle

```diff
@@ -95,6 +95,7 @@ dependencies {
     testImplementation(libraries.tomcatJdbc)
 
     testImplementation(libraries.jsonPathAssert)
+    testImplementation('com.google.guava:guava-testlib:30.1.1-jre')
```

Comment:
Hi @strehle  - Steve did this work on the fix as part of a contract with cloud.gov.  At this time that contract is done so Steve is unable to respond on behalf of cloud.gov to your questions so i'm going to do the best I can with what I know.

The most likely reason for the one test library being a higher rev then the spring deps was it is the latest version of that dependency and in general when the cloud.gov team works on an issue we try to update what we can to the latest and greatest to reduce the risk of having open issues/CVEs on code bases.

As far the fix path - again I can't speak for Steve as to why he choose the path for the fix he did vs your question on ExpiringUrlCache.  At the end of the day it solved the issue in what he probably felt was the best method based on his exp and the issue at hand.

At this point is there any reason this can not be approved and merged into the code base for a future release?  The issue at hand has caused some login outages on the cloud.gov platform when other IDP providers outside of cloud.gov's control have had outages to their IDP systems causing uaa to go into a crash loop.

Thanks

Reply:
please pull uaa develop branch into yours and then you can omit this version but only use 
testImplementation('com.google.guava:guava-testlib')